examples/infra/modDA.rs

```diff
@@ -0,0 +1,775 @@
+                    da_wait_between_polls,
+                    pub_key.clone(),
+                    known_nodes.clone(),
+                    true,
+                )
+                .into(),
+            );
+
+        let view_sync_network: WebCommChannel<
+            TYPES,
+            NODE,
+            ViewSyncCertificate<TYPES>,
+            ViewSyncVote<TYPES>,
+            MEMBERSHIP,
+        > = WebCommChannel::new(
+            WebServerNetwork::create(
+                &host.to_string(),
+                DEFAULT_WEB_SERVER_VIEW_SYNC_PORT,
+                wait_between_polls,
+                pub_key,
+                known_nodes.clone(),
+                false,
+            )
+            .into(),
+        );
+
+        WebServerDARun {
+            config,
+            quorum_network,
+            da_network,
+            view_sync_network,
+        }
+    }
+
+    fn get_da_network(
+        &self,
+    ) -> WebCommChannel<TYPES, NODE, DAProposal<TYPES>, DAVote<TYPES>, MEMBERSHIP> {
+        self.da_network.clone()
+    }
+
+    fn get_quorum_network(
+        &self,
+    ) -> WebCommChannel<
+        TYPES,
+        NODE,
+        QuorumProposal<TYPES, SequencingLeaf<TYPES>>,
+        QuorumVote<TYPES, SequencingLeaf<TYPES>>,
+        MEMBERSHIP,
+    > {
+        self.quorum_network.clone()
+    }
+
+    fn get_view_sync_network(
+        &self,
+    ) -> WebCommChannel<TYPES, NODE, ViewSyncCertificate<TYPES>, ViewSyncVote<TYPES>, MEMBERSHIP>
+    {
+        self.view_sync_network.clone()
+    }
+
+    fn get_config(&self) -> NetworkConfig<TYPES::SignatureKey, TYPES::ElectionConfigType> {
+        self.config.clone()
+    }
+}
+
+/// Main entry point for validators
+pub async fn main_entry_point<
+    TYPES: NodeType<ConsensusType = SequencingConsensus, Time = ViewNumber>,
+    MEMBERSHIP: Membership<TYPES> + Debug,
+    DANETWORK: CommunicationChannel<
+            TYPES,
+            Message<TYPES, NODE>,
+            DAProposal<TYPES>,
+            DAVote<TYPES>,
+            MEMBERSHIP,
+        > + Debug,
+    QUORUMNETWORK: CommunicationChannel<
+            TYPES,
+            Message<TYPES, NODE>,
+            QuorumProposal<TYPES, SequencingLeaf<TYPES>>,
+            QuorumVote<TYPES, SequencingLeaf<TYPES>>,
+            MEMBERSHIP,
+        > + Debug,
+    VIEWSYNCNETWORK: CommunicationChannel<
+            TYPES,
+            Message<TYPES, NODE>,
+            ViewSyncCertificate<TYPES>,
+            ViewSyncVote<TYPES>,
+            MEMBERSHIP,
+        > + Debug,
+    NODE: NodeImplementation<
+        TYPES,
+        Leaf = SequencingLeaf<TYPES>,
+        Exchanges = SequencingExchanges<
+            TYPES,
+            Message<TYPES, NODE>,
+            QuorumExchange<
+                TYPES,
+                SequencingLeaf<TYPES>,
+                QuorumProposal<TYPES, SequencingLeaf<TYPES>>,
+                MEMBERSHIP,
+                QUORUMNETWORK,
+                Message<TYPES, NODE>,
```

Comment:
these are some crazy trait bounds! I wish there was an easier way to do this.

Reply:
I did try to not have to include the VIEWSYNCNETWORK as one of the bounds, but it was going to take a lot longer and we wanted to get examples running. I could make a ticket to look into reducing the bounds for the next time we have spare cycles?

Reply:
I think we've devoted significant effort to this, and am not sure it's possible. +1 for making an issue, but I'm not sure how we could simplify bounds, sadly.